### DISCUSSION TITLE
how to keep socket connection alive?

### DISCUSSION THREAD

**Discussion by sibelius** [enisdenjo/graphql-ws#495] [General]
Every time we add or remove a subscription the connection is opened and closed

**Comment by enisdenjo** [comment]
Can you share your code example? Are you sure you're using one client and not more?

Also, if using [`lazy` mode on the client](https://the-guild.dev/graphql/ws/docs/interfaces/client.ClientOptions#lazy) (default), the connection will be established on-demand. In case you have just one subscription happening at a time and you want to keep the connection open, you have two options:

1. [Disable the `lazy` mode](https://the-guild.dev/graphql/ws/docs/interfaces/client.ClientOptions#lazy), this will have the client establish and maintain a connection as soon as it's instantiated
2. [Set a `lazyCloseTimeout`](https://the-guild.dev/graphql/ws/docs/interfaces/client.ClientOptions#lazyclosetimeout), this will debounce the disconnection after the last subscriber unsubscribes
